src/NuGet.Core/NuGet.ProjectManagement/Projects/BuildIntegratedNuGetProject.cs

```diff
@@ -54,6 +49,22 @@ public abstract class BuildIntegratedNuGetProject : NuGetProject, INuGetIntegrat
             INuGetProjectContext projectContext,
             bool throwOnFailure);
 
+        public abstract Task<bool> InstallPackageAsync(
+            PackageIdentity packageIdentity,
+            INuGetProjectContext nuGetProjectContext,
+            IEnumerable<NuGetFramework> successfulFrameworks,
+            IEnumerable<NuGetFramework> unsucessfulFrameworks,
```

Comment:
Yikes, that would take as-casting during the execute step. Sure, I'll add the overload.